src/qutipinterface.jl

```diff
@@ -0,0 +1,195 @@
+        evolved_state[i] = psi
+        i = i + 1
+    end
+    return time, evolved_state
+end
+
+mesolve(ME::QuLindbladMasterEq, rho0::QuBase.AbstractQuMatrix, tlist; solver=QuExpmV()) = mesolve(ME.hamiltonian, rho0, tlist, ME.collapse_ops; solver=solver)
+
+function mesolve(H::QuBase.AbstractQuArray, rho0::QuBase.AbstractQuMatrix, tlist, c_ops::Vector, eops::Vector; solver=QuExpmV())
+    expectation = zeros(Complex128, length(eops), length(tlist)-1)
+    for i=1:length(eops)
+        j = 1
+        for (t,psi) in QuStateEvolution(H, c_ops, rho0, tlist, solver)
+            expectation[i,j] = trace(coeffs(psi*eops[i]))
+            j = j + 1
+        end
+    end
+    return collect(tlist)[2:end], expectation
+end
+
+mesolve(ME::QuLindbladMasterEq, rho0::QuBase.AbstractQuMatrix, tlist, eops::Vector; solver=QuExpmV()) = mesolve(ME.hamiltonian, rho0, tlist, ME.collapse_ops, eops; solver=solver)
+
+
+@doc """
+Interface for `mcsolve`[http://qutip.org/docs/3.1.0/apidoc/functions.html#module-qutip.mcsolve] of QuTiP in QuDynamics
+
+`mcsolve` in QuDynamics, extended version.
+
+### Arguments
+
+Inputs :
+* H/ME :: AbstractQuArray/QuLindbladMasterEq Types
+
+  This parameter supports Hamiltonian of the system, Lindblad Master Equation
+* psi :: AbstractQuArray
+
+  Initial state vector of the system
+* tlist
+
+  Time step array/range
+* e_ops
+
+  Operators whose expectation values are to computed in the evolved states
+* solver
+
+  Keyword for selecting the solver from the various solvers supported by QuDynamics
+
+Output :
+* Array of tuples, which specify the time-step and the corresponding evolved state
+""" ->
+function mcsolve(H::QuBase.AbstractQuArray, psi0::QuBase.AbstractQuArray, tlist, c_ops::Vector; ntraj=1000)
+    time = collect(tlist)[2:end]
+    qumcwfen = QuMCWFEnsemble(complex(psi0), ntraj)
+    rho = complex(psi0*psi0')
+    rhos = Array(typeof(rho), length(tlist)-1)
+    for i=1:length(tlist)-1
+        rhos[i] = complex(zeros(rho))
+    end
+    for psi0 in qumcwfen
+        i = 1
+        for (t,psi) in QuStateEvolution(H, c_ops, psi0, tlist, QuMCWF())
+            rhos[i] = rhos[i] + (psi*psi')/length(qumcwfen)/norm(psi)^2
+            i = i + 1
+        end
+    end
+    return time, rhos
+end
+
+mcsolve(ME::QuLindbladMasterEq, psi0::QuBase.AbstractQuArray, tlist; ntraj=1000) = mcsolve(ME.hamiltonian, psi0, tlist, ME.collapse_ops; ntraj=ntraj)
+
+function eops_helper(eops, evolved_states)
+    expectation = zeros(Complex128, length(eops), length(evolved_states))
+    for i = 1:length(eops)
+        for j=1:length(evolved_states)
+            expectation[i,j] = trace(coeffs(evolved_states[j]*eops[i]))
+        end
+    end
+    return expectation
+end
+
+function mcsolve(H::QuBase.AbstractQuArray, psi0::QuBase.AbstractQuArray, tlist, c_ops::Vector, eops::Vector; ntraj=1000)
+    evolved_states = mcsolve(H, psi0, tlist, c_ops; ntraj=ntraj)[2]
+    expectation = eops_helper(eops, evolved_states)
```

Comment:
Here it is the same thing as above: you don't want to store all states (possible large matrices) and then do the expectation values (just numbers). You have to do the propagation as in `mcsolve` and calculate and store the expectation values in the inner loop ...